### PR TITLE
[Validator] Add docs for number constraints

### PR DESCRIPTION
**Pull request by jschaedl**
Feature PR: https://github.com/symfony/symfony/pull/28637

- [x] Positive
- [x] PositiveOrZero
- [x] Negative
- [x] NegativeOrZero

### Todo:

- [x] find a better example for `NegativeOrZero` constraint

### CODE REVIEW

File: reference/constraints/Negative.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The following constraints ensure that:
+
+* the ``withdraw`` of a  bankaccount ``TransferItem`` is a negative number (lesser than zero)
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Entity/Freezer.php
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]: Please rename it to TransferItem

File: reference/constraints/Negative.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class TransferItem
+        {
+
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]: Please remove the blank line

File: reference/constraints/Negative.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class TransferItem
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('withdraw', new Assert\Negative;
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]: Please add () after Negative

File: reference/constraints/Negative.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
+Negative
+========
+
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]: Please add a versionadded directive:

.. versionadded:: 4.3

File: reference/constraints/Negative.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            - `message`_
+            - `payload`_
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Negative`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\LesserThanValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+The following constraints ensure that:
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]: “The following constraint ensures that:”

It’s singular, isn’t it?

**Review by OskarStark — CHANGES_REQUESTED** [review]
Thank you 👍🏻

I left some comments.

File: reference/constraints/Negative.rst (hunk 5 above)
Comment (jschaedl) [reply reference/constraints/Negative.rst]: It is. Fixed.

File: reference/constraints/Negative.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class TransferItem
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('withdraw', new Assert\Negative();
```
Comment (OskarStark) [inline reference/constraints/Negative.rst]:
```suggestion
                $metadata->addPropertyConstraint('withdraw', new Assert\Negative());
```

File: reference/constraints/NegativeOrZero.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class TransferItem
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('withdraw', new Assert\NegativeOrZero();
```
Comment (OskarStark) [inline reference/constraints/NegativeOrZero.rst]:
```suggestion
                $metadata->addPropertyConstraint('withdraw', new Assert\NegativeOrZero());
```

File: reference/constraints/Positive.rst [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Employee
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('income', new Assert\Positive();
```
Comment (OskarStark) [inline reference/constraints/Positive.rst]:
```suggestion
                $metadata->addPropertyConstraint('income', new Assert\Positive());
```

File: reference/constraints/PositiveOrZero.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Person
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('siblings', new Assert\PositiveOrZero();
```
Comment (OskarStark) [inline reference/constraints/PositiveOrZero.rst]:
```suggestion
                $metadata->addPropertyConstraint('siblings', new Assert\PositiveOrZero());
```

File: reference/constraints/map.rst.inc [hunk 10]
```diff
@@ -38,6 +38,13 @@ Comparison Constraints
 * :doc:`Range </reference/constraints/Range>`
 * :doc:`DivisibleBy </reference/constraints/DivisibleBy>`
 
+Number Constraints
+~~~~~~~~~~~~~~~~~~
+* :doc:`Date </reference/constraints/Positive>`
```
Comment (OskarStark) [inline reference/constraints/map.rst.inc]:
```suggestion
* :doc:`Positive </reference/constraints/Positive>`
```

File: reference/constraints/map.rst.inc [hunk 11]
```diff
@@ -38,6 +38,13 @@ Comparison Constraints
...
+* :doc:`Date </reference/constraints/PositiveOrZero>`
```
Comment (OskarStark) [inline reference/constraints/map.rst.inc]:
```suggestion
* :doc:`PositiveOrZero </reference/constraints/PositiveOrZero>`
```

File: reference/constraints/map.rst.inc [hunk 12]
```diff
@@ -38,6 +38,13 @@ Comparison Constraints
...
+* :doc:`Date </reference/constraints/Negative>`
```
Comment (OskarStark) [inline reference/constraints/map.rst.inc]:
```suggestion
* :doc:`Negative </reference/constraints/Negative>`
```

File: reference/constraints/map.rst.inc [hunk 13]
```diff
@@ -38,6 +38,13 @@ Comparison Constraints
...
+* :doc:`Date </reference/constraints/NegativeOrZero>`
```
Comment (OskarStark) [inline reference/constraints/map.rst.inc]:
```suggestion
* :doc:`NegativeOrZero </reference/constraints/NegativeOrZero>`
```

File: reference/constraints/NegativeOrZero.rst [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\NegativeOrZero`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\LesserThanOrEqualValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+The following constraint ensure that:
+
+* the ``withdraw`` of a  bankaccount ``TransferItem`` is a negative number or equal to zero
```
Comment (OskarStark) [inline reference/constraints/NegativeOrZero.rst]: I propose to rename `TransferItem` to `BankAccount`, but to be honest, a withdraw where 0 is allowed sounds strange 🤔 Maybe we could find a better example.

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: reference/constraints/NegativeOrZero.rst [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class TransferItem
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('withdraw', new Assert\NegativeOrZero();
+            }
+        }
```
Comment (OskarStark) [inline reference/constraints/NegativeOrZero.rst]:
```suggestion
        }

Available Options
-----------------

.. include:: /reference/constraints/_groups-option.rst.inc

message
~~~~~~~

**type**: ``string`` **default**: ``This is not a valid Business Identifier Code (BIC).``

The default message supplied when the value does not pass the BIC check.

You can use the following parameters in this message:

===============  ==============================================================
Parameter        Description
===============  ==============================================================
``{{ value }}``  The current (invalid) BIC value
===============  ==============================================================

.. include:: /reference/constraints/_payload-option.rst.inc
```

Please add the available options at the bottom (I added a suggestion, but copied the part from the [Bic](https://github.com/symfony/symfony-docs/blob/master/reference/constraints/Bic.rst) constraint, so please adjust it)

**Comment by OskarStark** [comment]
Please include your new constraints in [reference/constraints.rst](https://github.com/symfony/symfony-docs/blob/master/reference/constraints.rst), otherwise Sphinx won't be able to build the docs.

File: reference/constraints/Negative.rst [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+
+The default message supplied when the value is not less than zero.
+
+You can use the following parameters in this message:
+
+=============================  ================================================
+Parameter                      Description
+=============================  ================================================
+``{{ compared_value }}``       Always zero
+``{{ compared_value_type }}``  The expected value type
```
Comment (jschaedl) [inline reference/constraints/Negative.rst]: Not sure if we should include `compared_value` and `compared_value_type` here. They are available, but it doesn't make so much sense to me to use them. WDYT?

File: reference/constraints/NegativeOrZero.rst [hunk 17]
```diff
@@ -24,7 +24,7 @@ Basic Usage
 
 The following constraint ensure that:
 
-* the ``withdraw`` of a  bankaccount ``TransferItem`` is a negative number or equal to zero
+* the ``level`` of a ``UnderGroundGarage`` is a negative number or equal to zero
```
Comment (jschaedl) [inline reference/constraints/NegativeOrZero.rst]: @OskarStark WDYT?

Comment (OskarStark) [reply reference/constraints/NegativeOrZero.rst]: Nice idea 👍

**Review by OskarStark — APPROVED** [review]
Great work, thank you! 👍

File: reference/constraints/Negative.rst (hunk 16 above)
Comment (OskarStark) [reply reference/constraints/Negative.rst]: I am not sure if we haven't added such (if available) parameters for other constraints.

Please help @xabbuh @javiereguiluz

Comment (javiereguiluz) [reply reference/constraints/Negative.rst]: Even if they won't be used frequently, for consistency with the other constraints I'd prefer to keep them. We always try to document all parameters. Thanks.

**Comment by javiereguiluz** [comment]
@jschaedl thanks a lot for providing these docs (and the related feature too). Congrats on your first (great) Symfony Docs contribution!

**Pull request closed without merging**